### DISCUSSION TITLE
How to change language application to spanish?

### DISCUSSION THREAD

**Discussion by drfcozapata** [Murmele/Gittyup#793] [General]
Looking at the application, it's supposed that by changing Tools > Options > General > Language it should be possible to change it to Spanish. However it doesn't do it.

I read the documentation and it explains that it is possible to change the language to spanish with the command `LANG=<lang> <executable>`, so I edited the .desktop file and changed the line `Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gittyup com. github.Murmele.Gittyup` to `Exec=env LANG=en /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gittyup com.github.Murmele.Gittyup`. However, it doesn't change to Spanish either.

How can I make the language change? Thanks in advance.

**Comment by safetycar** [comment]
Over the language selection dropbox, there is a "No translation" checkbox, maybe make sure it is disabled.

**Reply by drfcozapata** [reply]
It's disabled. It still does not change the language.

**Comment by alerque** [comment]
```diff
-Exec=env LANG=en /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gittyup com.github.Murmele.Gittyup
+Exec=env LANG=es /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gittyup com.github.Murmele.Gittyup
```

`en`→`es`

If correcting that mistake doesn't help then something about flatpack is not passing the environment through. You may need to try something like `--command="env LANG=es gittyup"` or something like that. Also there is no telling if the Flatpack container has locales enabled at all, it may well not have any system locales even enabled in which case gittyup on it's own won't be able to change it at all.

**Reply by drfcozapata** [reply]
My system's locales are as follows:
LANG=en_VE.UTF-8
LC_CTYPE="es_VE.UTF-8”
LC_NUMERIC=en_VE.UTF-8
LC_TIME=en_VE.UTF-8
LC_COLLATE="es_VE.UTF-8”
LC_MONETARY=en_VE.UTF-8
LC_MESSAGES="en_VE.UTF-8”
LC_PAPER=en_VE.UTF-8
LC_NAME=en_VE.UTF-8
LC_ADDRESS=en_VE.UTF-8
LC_TELEPHONE=en_VE.UTF-8
LC_MEASUREMENT=en_VE.UTF-8
LC_IDENTIFICATION=en_VE.UTF-8
LC_ALL=

I checked the locale settings of the flatpak container and it does not have:
[Context]
shared=network;ipc;
sockets=x11;wayland;fallback-x11;ssh-auth;
devices=dri;
filesystems=home;/tmp;xdg-config/kdeglobals:ro;xdg-run/keyring;
[Session Bus Policy]
com.canonical.AppMenu.Registrar=talk
org.kde.kconfig.notify=talk
org.kde.KGlobalSettings=talk
org.freedesktop.Flatpak=talk
org.freedesktop.Notifications=talk
org.freedesktop.secrets=talk

I tried making the change you suggested in the .desktop file (+Exec=env LANG=en /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gittyup com.github.Murmele.Gittyup) and it didn't work. 

Anyway, thanks for your time.

**Reply by alerque** [reply]
> My system's locales are as follows:

Yes, but *your system* locale isn't very relevant here, it only really matters what the system locale is inside the Flatpack (which is basically a containerized system of its own).

Do you not have the option of running a version that isn't containerized? Everything is layer and layers more complicated to debug when you're dealing with a containerized app system.
